ribbon-core/src/main/java/com/netflix/client/config/DynamicPropertyRepository.java

```diff
@@ -0,0 +1,61 @@
+package com.netflix.client.config;
+
+import java.lang.reflect.Type;
+import java.util.Comparator;
+import java.util.ServiceLoader;
+import java.util.function.Consumer;
+import java.util.stream.StreamSupport;
+
+/**
+ * Ribbon specific abstraction to a Repository of dynamic configuration properties
+ */
+public interface DynamicPropertyRepository {
+    DynamicPropertyRepository DEFAULT = findDynamicPropertyRepository();
+
+    /**
+     * Proprity of the repository when loading from the server loader.  The highest priority wins
+     * @return
+     */
+    int getPriority();
+
+    /**
+     * Look for a static DynamicPropertyRepository via the service loader.  It's OK to have an empty repository for
+     * situations where one is provided via dependency injection.
+     * @return
+     */
+    static DynamicPropertyRepository findDynamicPropertyRepository() {
+        return StreamSupport.stream(ServiceLoader.load(DynamicPropertyRepository.class).spliterator(), false)
+                .sorted(Comparator
+                        .comparingInt(DynamicPropertyRepository::getPriority)
+                        .thenComparing(Comparator.comparing(f -> f.getClass().getCanonicalName())))
+                .findFirst()
+                .orElse(null);
```

Comment:
Is the default a null obj or should there be an NoOp version that can be referenced?